src/components/forms/CreateRiddleForm.tsx

```diff
@@ -0,0 +1,45 @@
+import { FormContainer, useForm } from 'react-hook-form-mui';
+import { useCallback } from 'react';
+import { Button, Stack } from '@mui/material';
+import { Typography } from '@mui/material/styles/createTypography';
+
+import type { RiddleUpsertDetail } from '../../utils/Types';
+
+import { TextFieldFormComponent } from './generic/TextFieldFormComponent';
```

Comment:
Nevěděl jsem úplně, jak ty formulářové komponenty strukturovat.
Napadlo mě buď složky `generic` a  `implemented` (to, co je nyní, jen ty implementace nejsou ve složce), nebo případně to rozdělit podle komponent, tzn. mít složky `Autocomplete`, `RadioButton` atd., kde budou implementace i generčtější verze.